### PR TITLE
Flatten QueryPhaseResultConsumer

### PR DESCRIPTION
**Pull request by original-brownbear**
This is a trivial change with whitespaces ignored :) 

No need for a separate PendingMerges nested class, this thing has a lot of state, lets accept that (for now) and keep it simple. Having a redundant nested class already caused a bug in #116171 and just adds overhead both in runtime and when reading the code.

### CODE REVIEW

**Review by javanna — APPROVED** [review]
I agree with the rationale behind the change. Inner classes (non static nested classes) are hard to follow. I trust that you moved all the code carefully, the diff is impossible to read :)

**Review by benwtrent — APPROVED** [review]

**Comment by original-brownbear** [comment]
Thanks both!
